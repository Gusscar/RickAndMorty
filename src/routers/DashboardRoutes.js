import React from "react";

import { Routes, Route } from "react-router-dom";

import { Navbar } from "../ui/NavBar";
import { MortyScreen } from "../components/morty/MortyScreen";
import { SearchScreen } from "../components/search/SearchScreen";
import { CharactersScreen } from "../components/characters/CharactersScreen";
import { LocationScreen } from "../components/location/LocationScreen";

export const DashboardRoutes = () => {
  const containerStyle = {
    backgroundImage:
      "url(https://img.thedailybeast.com/image/upload/c_crop,d_placeholder_euli9k,h_1687,w_3000,x_0,y_0/dpr_1.5/c_limit,w_1600/fl_lossy,q_auto/v1573446366/191107-leon-rick-morty-tease_nmv5um)",
    width: "100vp",
    height: "100vh",
    backgroundRepeat: "no-repeat",
    backgroundSize: "cover",
  };
  return (
    <div style={containerStyle}>
      <Navbar />
      <div style={{ maxWidth: "100%" }}>
        <Routes>
          <Route path="/location" element={<LocationScreen />} />
          <Route path="/characters" element={<MortyScreen />} />
          <Route path="/search" element={<SearchScreen />} />
          <Route path="characters/:id" element={<CharactersScreen />} />

          <Route path="/" element={<MortyScreen />} />
        </Routes>
      </div>
    </div>
  );
};
