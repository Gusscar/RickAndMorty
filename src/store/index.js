import { configureStore } from "@reduxjs/toolkit";

//reducer
import characters from "./slices/characters";
import locations from "./slices/location";

export default configureStore({
  reducer: {
    characters,
    locations,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});
