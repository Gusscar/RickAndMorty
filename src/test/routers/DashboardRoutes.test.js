import { mount } from "enzyme";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../auth/authContext";
import { DashboardRoutes } from "../../routers/DashboardRoutes";
import store from "../../store";

describe("Test in <DashboardRoutes/>", () => {
  const contextValue = {
    user: {
      logged: true,
      name: "Santiago",
    },
  };

  test("should Show correctly", () => {
    const wrapper = mount(
      <Provider store={store}>
        <AuthContext.Provider value={contextValue}>
          <MemoryRouter initialEntries={["/"]}>
            <DashboardRoutes />
          </MemoryRouter>
        </AuthContext.Provider>
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".text-info").text().trim()).toBe("Santiago");
  });

  test("should Show correctly Location", () => {
    const wrapper = mount(
      <Provider store={store}>
        <AuthContext.Provider value={contextValue}>
          <MemoryRouter initialEntries={["/location"]}>
            <DashboardRoutes />
          </MemoryRouter>
        </AuthContext.Provider>
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("div").exists()).toBe(true);
  });
});
