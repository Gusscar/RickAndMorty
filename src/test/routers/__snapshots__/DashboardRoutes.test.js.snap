// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <DashboardRoutes/> should Show correctly 1`] = `
<div
  style={
    Object {
      "backgroundImage": "url(https://img.thedailybeast.com/image/upload/c_crop,d_placeholder_euli9k,h_1687,w_3000,x_0,y_0/dpr_1.5/c_limit,w_1600/fl_lossy,q_auto/v1573446366/191107-leon-rick-morty-tease_nmv5um)",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
      "height": "100vh",
      "width": "100vp",
    }
  }
>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Rick and Morty
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/location"
          onClick={[Function]}
        >
          Locations
        </a>
        <a
          className="nav-item nav-link"
          href="/characters"
          onClick={[Function]}
        >
          Characters
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Santiago
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
          to="/login"
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>
  <div
    className="container"
  >
    <div
      style={
        Object {
          "marginTop": "50px",
        }
      }
    >
      <div
        className="row rows-cols-1 row-cols-md-3 g-3 animate__animated animate__fadeIn"
      />
      <nav
        style={
          Object {
            "marginBottom": "10px",
            "marginTop": "20px",
          }
        }
      >
        <ul
          className="pagination justify-content-center"
        />
      </nav>
    </div>
  </div>
</div>
`;

exports[`Test in <DashboardRoutes/> should Show correctly Location 1`] = `
<div
  style={
    Object {
      "backgroundImage": "url(https://img.thedailybeast.com/image/upload/c_crop,d_placeholder_euli9k,h_1687,w_3000,x_0,y_0/dpr_1.5/c_limit,w_1600/fl_lossy,q_auto/v1573446366/191107-leon-rick-morty-tease_nmv5um)",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
      "height": "100vh",
      "width": "100vp",
    }
  }
>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Rick and Morty
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/location"
          onClick={[Function]}
        >
          Locations
        </a>
        <a
          className="nav-item nav-link"
          href="/characters"
          onClick={[Function]}
        >
          Characters
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Santiago
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
          to="/login"
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>
  <div
    className="container"
  >
    <div
      className="row"
    />
  </div>
</div>
`;
